src/main.ts

```diff
@@ -1867,15 +1925,20 @@ function generateUnwrap(ctx: Context, fullProtoTypeName: string, fieldNames: Str
       chunks.push(code`unwrap(message: Struct): {[key: string]: any} {
         const object: { [key: string]: any } = {};
         [...message.fields.keys()].forEach((key) => {
-          object[key] = message.fields.get(key);
+          const unwrappedValue = Value.unwrap(message.fields.get(key));
+          object[key] = unwrappedValue !== undefined ? unwrappedValue : message.fields.get(key);
```

Comment:
Question: Just curious, but instead of `: message.fields.get(key)`, should this be just `: undefined`? Like if `Value.unwrap` can't unwrap the value, is there really a potentially valid-but-not-unwrappable value we need to get?

Reply:
Can't recall why I put it there but there was something - I'll look into what happens if we remove it

Reply:
The reason we have that here is for the case where the result of `message.fields.get(key)` is not necessarily actually wrapped and this cause the result of `Value.unwrap()` to return `undefined`

Returning the actual value instead help us avoid these. 
Removing this broke the tests

Reply:
Ah okay, happy to keep it then.

Out of curiosity, is it expected that we'll end up with not-actually-wrapped values in should-be-wrapped cases, or was an existing test just being particularly ocd?

Reply:
The second part :)

Reply:
Hm, I think I realized what was happening ... 

In the `findOneHero` test, the server was only invoking `Struct.wrap(...)` once, but never invoking `ListValue.wrap` or `Value.wrap` on any of the inner contents.

Even after adding wrapper entries for them:

```
wrappers[".google.protobuf.Struct"] = { fromObject: Struct.wrap, toObject: Struct.unwrap } as any;
wrappers[".google.protobuf.Value"] = { fromObject: Value.wrap, toObject: Value.unwrap } as any;
wrappers[".google.protobuf.ListValue"] = { fromObject: ListValue.wrap, toObject: ListValue.unwrap } as any;
```

`Value.wrap` / `ListValue.wrap` are never invoked.

So I think that's why, inside the client `.unwrap`s, we saw things that were "hey, wait a sec, not wrapped".

I was able to fix this by making `Struct.wrap` deeply wrap, and `Struct.unwrap` deeply unwrap...which AFAICT fixed the issue.

But it broke ts-proto's non-NestJS `Struct.encode` / `Struct.decode` methods, which specifically assumed that `wrap`/`unwrap` were shallow.

So now I need to go back and teach the `encode`/`decode` methods about the deepness of wrap/unwrap...or maybe for now we just have two versions of wrap/unwrap; one that is deep for NestJS, and one that is shallow for not-NestJS...

I'm going to go ahead and push up the deep-wrap/unwrap changes up to this PR, which will break some non-NestJS tests, but AFAICT that's the right path forward for NestJS's Struct support.

Does that seem right?

Reply:
I agree that deeply wrapping and unwrapping is the way to go

Don't want to break the tests.

Reply:
Well, I cheated and just copy/pasted both deep & shallow versions of the wrappers, but it seems to be working for now. :-)

I really haven't personally used the Struct/List/ListValue types, and helping on this PR was the closest I've been to the code... I feel like something is still a little off...like how in the regular/non-NestJs output, `Struct.encode` takes a `Struct` but it should take an `any`...

Anyway, I'm going to punt on that for now. Tests are passing so I'll merge and :crossed_fingers: seems like it should work.

Thanks for the PR!